monitoring/honeybadger_deployment.py

```diff
@@ -0,0 +1,131 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: honeybadger_deployment
+author: Benjamin Curtis
+short_description: Notify Honeybadger.io about app deployments
+description:
+   - Notify Honeybadger.io about app deployments (see http://docs.honeybadger.io/article/188-deployment-tracking)
+options:
+  token:
+    description:
+      - API token.
+    required: true
+  environment:
+    description:
+      - The environment name, typically 'production', 'staging', etc.
+    required: true
+  user:
+    description:
+      - The username of the person doing the deployment
+    required: false
+  repo:
+    description:
+      - URL of the project repository
+    required: false
+  revision:
+    description:
+      - A hash, number, tag, or other identifier showing what revision was deployed
+    required: false
+  url:
+    description:
+      - Optional URL to submit the notification to.
+    required: false
+    default: "https://api.honeybadger.io/v1/deploys"
+  validate_certs:
+    description:
+      - If C(no), SSL certificates for the target url will not be validated. This should only be used
+        on personally controlled sites using self-signed certificates.
+    required: false
+    default: 'yes'
+    choices: ['yes', 'no']
+
+# informational: requirements for nodes
+requirements: [ urllib, urllib2 ]
+'''
+
+EXAMPLES = '''
+- honeybadger_deployment: token=AAAAAA
+                          environment='staging'
+                          user='ansible'
+                          revision=b6826b8
+                          repo=git@github.com:user/repo.git
+'''
+
+# ===========================================
+# Module execution.
+#
+
+def main():
+
+    module = AnsibleModule(
+        argument_spec=dict(
+            token=dict(required=True),
+            environment=dict(required=True),
+            user=dict(required=False),
+            repo=dict(required=False),
+            revision=dict(required=False),
+            url=dict(required=False, default='https://api.honeybadger.io/v1/deploys'),
+            validate_certs=dict(default='yes', type='bool'),
+        ),
+        supports_check_mode=True
+    )
+
+    params = {}
+
+    if module.params["environment"]:
+        params["deploy[environment]"] = module.params["environment"]
+
+    if module.params["user"]:
+        params["deploy[local_username]"] = module.params["user"]
+
+    if module.params["repo"]:
+        params["deploy[repository]"] = module.params["repo"]
+
+    if module.params["revision"]:
+        params["deploy[revision]"] = module.params["revision"]
+
+    params["api_key"] = module.params["token"]
+
+    url = module.params.get('url')
+
+    # If we're in check mode, just exit pretending like we succeeded
+    if module.check_mode:
+        module.exit_json(changed=True)
+
+    try:
+        data = urllib.urlencode(params)
```

Comment:
This should be handled by fetch_url and should be unnecessary, in which case you should be able to ditch the urllib requirement as well -- but worth double-checking to be sure.